### PR TITLE
get_repositories for toolshed category

### PR DESCRIPTION
**Pull request by cat-bro**
Add method for toolshed GET categories/<category_id>/repositories

### CODE REVIEW

File: bioblend/toolshed/categories/__init__.py [hunk 1]
```diff
@@ -45,3 +45,71 @@ def show_category(self, category_id):
         :return: details of the given category
         """
         return self._get(id=category_id)
+
+    def get_repositories(self, category_id, sort_key=None, sort_order=None):
```
Comment (simonbray) [inline bioblend/toolshed/categories/__init__.py:49]:
```suggestion
    def get_repositories(self, category_id, sort_key='name', sort_order='asc'):
```

File: bioblend/toolshed/categories/__init__.py [hunk 2]
```diff
@@ -45,3 +45,71 @@ def show_category(self, category_id):
...
+        """
+        Returns a dictionary of information for a repository category including
+        a list of repositories belonging to the category
```
Comment (simonbray) [inline bioblend/toolshed/categories/__init__.py:52]:
```suggestion
        a list of repositories belonging to the category.
```

File: bioblend/toolshed/categories/__init__.py [hunk 3]
```diff
@@ -45,3 +45,71 @@ def show_category(self, category_id):
...
+        :param sort_key: key for sorting. Options are 'name' or 'owner' (default 'name').
+
+        :type  sort_order: str
+        :param sort_order: ordering of sorted output. Options are 'asc' or 'desc' (default 'asc').
+
+        :rtype: dict
+        :return: A dict containing information about the category
+          including a list of repository dicts
+          For example::
+          {'deleted': False,
```
Comment (simonbray) [inline bioblend/toolshed/categories/__init__.py:67]:
```suggestion
        :return: A dict containing information about the category
          including a list of repository dicts.
          For example::

          {'deleted': False,
```

**Review by simonbray — COMMENTED** [review]
I was a bit surprised there are no tests for the part of BioBlend relating to the toolshed, but I suppose, as that's the case, it's fine to skip them for this PR as well. I tested locally and it looks fine to me.

**Comment by cat-bro** [comment]
Thanks @simonbray.  I'm not sure how I'd go about writing a test for this.  With bioblend.galaxy the testing process spins up a local galaxy, I'm not sure if there's an easy way to establish a local toolshed for testing.  I guess there could be tests that use the main toolshed and check for a 200 response for each function, but then if the toolshed were ever offline it would interfere with the CI process.

**Review by simonbray — APPROVED** [review]

File: bioblend/toolshed/categories/__init__.py [hunk 4]
```diff
@@ -45,3 +45,72 @@ def show_category(self, category_id):
...
+          'url': '/api/categories/589548af7e391bcf'}
+        """
+
+        params = {}
+        if sort_key:
+            params.update({'sort_key': sort_key})
+        if sort_order:
+            params.update({'sort_order': sort_order})
+
+        url = self._make_url() + f'/{category_id}/repositories'
```
Comment (nsoranzo) [inline bioblend/toolshed/categories/__init__.py:115]:
```suggestion
        url = self._make_url(category_id) + '/repositories'
```

File: bioblend/toolshed/categories/__init__.py [hunk 5]
```diff
@@ -45,3 +45,72 @@ def show_category(self, category_id):
...
+                            'owner': 'iuc',
+                            'private': False,
+                            'remote_repository_url': 'https://github.com/gregvonkuster/galaxy-csg',
+                            'times_downloaded': 152,
+                            'type': 'unrestricted',
+                            'user_id': 'b563abc230aa8fd0'},
+          # ...
+          ],
+          'repository_count': 11,
+          'url': '/api/categories/589548af7e391bcf'}
```
Comment (nsoranzo) [inline bioblend/toolshed/categories/__init__.py:106]:
```suggestion
            {'deleted': False,
             'description': 'Tools for constructing and analyzing 3-dimensional shapes and '
                            'their properties',
             'id': '589548af7e391bcf',
             'model_class': 'Category',
             'name': 'Constructive Solid Geometry',
             'repositories': [{'create_time': '2016-08-23T18:53:23.845013',
                               'deleted': False,
                               'deprecated': False,
                               'description': 'Adds a surface field to a selected shape '
                                              'based on a given mathematical expression',
                               'homepage_url': 'https://github.com/gregvonkuster/galaxy-csg',
                               'id': 'af2ccc53697b064c',
                               'metadata': {'0:e12b55e960de': {'changeset_revision': 'e12b55e960de',
                                                               'downloadable': True,
                                                               'has_repository_dependencies': False,
                                                               'id': 'dfe022067783215f',
                                                               'includes_datatypes': False,
                                                               'includes_tool_dependencies': False,
                                                               'includes_tools': True,
                                                               'includes_tools_for_display_in_tool_panel': True,
                                                               'includes_workflows': False,
                                                               'malicious': False,
                                                               'missing_test_components': False,
                                                               'model_class': 'RepositoryMetadata',
                                                               'numeric_revision': 0,
                                                               'repository_id': 'af2ccc53697b064c'}},
                               'model_class': 'Repository',
                               'name': 'icqsol_add_surface_field_from_expression',
                               'owner': 'iuc',
                               'private': False,
                               'remote_repository_url': 'https://github.com/gregvonkuster/galaxy-csg',
                               'times_downloaded': 152,
                               'type': 'unrestricted',
                               'user_id': 'b563abc230aa8fd0'}],
             'repository_count': 11,
             'url': '/api/categories/589548af7e391bcf'}
```

**Review by nsoranzo — COMMENTED** [review]
> I'm not sure if there's an easy way to establish a local toolshed for testing. I guess there could be tests that use the main toolshed and check for a 200 response for each function, but then if the toolshed were ever offline it would interfere with the CI process.

I think it should be fine to test against the main or test ToolShed, and we could add a `skip_unless_toolshed` decorator to the test class if it proves unstable.
Any way, this shouldn't block this PR!

**Comment by cat-bro** [comment]
Changes committed, thanks @nsoranzo!

**Comment by nsoranzo** [comment]
Thanks @cat-bro !

**Pull request merged**